botocore/docs/bcdoc/restdoc.py

```diff
@@ -214,32 +225,38 @@ def flush_structure(self, docs_link=None):
         """
         # We are at the root flush the links at the beginning of the
         # document
-        if len(self.path) == 1:
+        path_length = len(self.path)
+        if path_length == 1:
             if self.hrefs:
                 self.style.new_paragraph()
                 for refname, link in self.hrefs.items():
                     self.style.link_target_definition(refname, link)
+        # Clear docs_link at the correct depth to prevent passing a non-related link.
+        elif path_length == SECTION_METHOD_PATH_DEPTH.get(self.path[1]):
+            docs_link = None
         value = self.getvalue()
         for name, section in self._structure.items():
             # Checks is the AWS API Documentation link has been generated.
             # If it has been generated, it gets passed as a the doc_link parameter.
-            if b'`AWS API Documentation <' in value:
-                start = value.find(b'`AWS API Documentation <')
-                end = value.find(b'>`_', start) + 3
-                docs_link = value[start:end] + b'\n\n'
+            match = re.search(DOCUMENTATION_LINK_REGEX, value.decode())
```

Comment:
After updating the regex string with `re.compile` to a regex object, you can update this to `DOCUMENTATION_LINK_REGEX.search(value.decode())`